kube/src/config/mod.rs

```diff
@@ -181,19 +181,9 @@ impl Config {
             timeout: Some(DEFAULT_TIMEOUT),
             accept_invalid_certs,
             identity: identity_pem.map(|i| (i, String::from(IDENTITY_PASSWORD))),
-            auth_header: Authentication::from_auth_info(&loader.user).await?,
+            auth_info: loader.user,
         })
     }
-
-    /// Get a valid HTTP `Authorization` header that can authenticate to the cluster
-    ///
-    /// Will renew tokens if required (and configured to).
-    ///
-    /// NOTE: This is `None` if the `Config` isn't configured to use token-based authentication
-    /// (such as anonymous access, or certificate-based authentication).
-    pub async fn get_auth_header(&self) -> Result<Option<header::HeaderValue>> {
```

Comment:
This is a breaking change. We can expose `Authentication` and add `to_header` method if necessary.

Reply:
I think this is fine, we have been restructuring a lot in this area for the better.